### ISSUE TITLE
Feature Request: Add one way sync

### ISSUE THREAD

**Issue by nikolayhg** [nextcloud/desktop#1090]
### Expected behaviour

The desktop client to have the following options:

- One way sync of  **a nextcloud folder to a local folder**:
  - Nextcloud files are synced/downloaded to the local folder.
  - Local files are **not** synced/uploaded to nextcloud.

- One way sync of **a local folder to a nextcloud folder**:
  - Local files are synced/uploaded to nextcloud.
  - Nextcloud files are **not** synced/downloaded to the local folder.

### Actual behaviour
only two way sync.

### Others
I could not find another feature request that would cover the expected behaviour. #267 mentions one way sync, but _without the need of a local copy_, which is not desirable for this feature request.

In the [nextcloud help](https://help.nextcloud.com/t/one-way-sync-feature/29126) one user claims that _Nextcloud normally does not keep backups of the files by design_. Would be great if the developers comment on this claim/issue.

**Comment by hack4mer** [comment]
Hi, this is much-needed feature. Any update?

**Comment by magnetophon** [comment]
The other way around would also be handy in certain cases: only sync from cloud to local, but not back.

**Comment by michaelstingl** [comment]
> The other way around would also be handy in certain cases: only sync from cloud to local, but not back.

Try with a folder that got shared with your user account, but without edit permissions.

**Comment by magnetophon** [comment]
@michaelstingl Sorry, I don't get what you mean.

**Comment by WhiteLionATX** [comment]
yes, one way sync is a much needed feature!

**Comment by l4n-socrates** [comment]
indeed, that would very usefull, definitely for backups

**Comment by dsiminiuk** [comment]
Mee too! Great idea. I have a use for this and found this thread!

PLEEEEEASE!

**Comment by JohnOrme** [comment]
Would love this!

**Comment by spacemanspiff2007** [comment]
Same here!

**Comment by meika** [comment]
Yes, I support this as an completely needed option, currently working around the lack of this but it is a nervous thing all round. I find the two way sync a lot less useful for my needs generally. In fact it is very annoying and regret spending any time on nextcloud at all.... looking for an exit strategy. Didn't know I was going to end up typing that.

**Comment by Windyo** [comment]
bump as well :)

**Comment by inakrin** [comment]
Need this feature too...

**Comment by Dzeri96** [comment]
I agree, this is a must-have. Currently I'm playing around with other webdav clients, but native support would be better.

**Comment by superzanti** [comment]
I really would love this feature!

**Comment by mzed2k** [comment]
This feature would make life easier! The WebDAV interface is no option for huge files and unfortunately rclone doesn't support NC chunked upload.

**Comment by Rixillo** [comment]
Bump!

**Comment by delbonis** [comment]
A use case that I'd like this for would be to sync the screenshots folders I have on my different devices to my main server without each client syncing down screenshots taken from other devices.  Not sure why this isn't possible without having to do it myself using something like rclone and WebDAV.

**Comment by RandieM** [comment]
This would indeed be a great feature, especially in cases where one needs to be sure that the files from the cloud will not be accidentally edited or deleted.

**Comment by ALERTua** [comment]
This feature is much needed.

**Comment by Durburz** [comment]
Yes, I need this feature too.

**Comment by szczot3k** [comment]
Here I am - bump! This feature is much needed.

**Comment by hanneschu1z** [comment]
Would be great!

**Comment by gotros** [comment]
Bump!

**Comment by dsiminiuk** [comment]
In my case I ended up mounting the user's webdav folder with davfs2 and autofs, and a bash script to copy the content to it.

**Comment by mzed2k** [comment]
Rclone is an option for the file transfer part, but for the added value of the full featured Nextcloud Desktop client, like folder watching, notifications,  etc. you still need to install it.

**Comment by FathiBenNasr** [comment]
> 
> 
> > The other way around would also be handy in certain cases: only sync from cloud to local, but not back.
> 
> Try with a folder that got shared with your user account, but without edit permissions.

That's a good idea for one way sync: NC to local sync.
I need the other only one wat sync from local to NC, as I use NC to backup a NAS shared folder.

**Comment by SharkIng** [comment]
>The other way around would also be handy in certain cases: only sync from cloud to local, but not back.

Sync from local to cloud is also a handy feature.
Would like to know if there is any read map for this.

Use case: you have tons of video, photo or ISO or whatever that's super large (few GB per file) that you would like to keep it sync, backup, and able to use it when needed or use it on another PC. You don't want it sync on certain PC or you don't want a copy on one PC to save some disk space. It's good to be able to see it as symlink and able to download single file when needed.

I believe Dropbox and iCloud have it already. Really useful feature.

**Comment by manasmgkar** [comment]
> > > The other way around would also be handy in certain cases: only sync from cloud to local, but not back.
> > 
> > 
> > Try with a folder that got shared with your user account, but without edit permissions.
> 
> That's a good idea for one way sync: NC to local sync. I need the other only one wat sync from local to NC, as I use NC to backup a NAS shared folder.

+1 for this, developers please implement something

**Comment by EsmailELBoBDev2** [comment]
featuer wanted from 2019 and we are literally in 2022 (happy new year btw) and still no updates? fuck i want this featuer, tell me there is hope

**Comment by szczot3k** [comment]
Dude... Did you really wait until new year's just to write this comment? Filip Kszczotmob.: +48 539 965 536email: ***@***.*** ---- On Sat, 01 Jan 2022 00:16:55 +0100  ***@***.***  wrote ----
featuer wanted from 2019 and we are literally in 2022 (happy new year btw) and still no updates? fuck i want this featuer, tell me there is hope

**Comment by EsmailELBoBDev2** [comment]
@szczot3k nope, it's a luck (good luck i would like to call it)
i wanted this featuer some months but did not know that lot's of other people want it too

**Comment by merrpink** [comment]
This is a basic feature. I ended up paying for another service that gave me the tools I need.

**Comment by EsmailELBoBDev2** [comment]
To be honest I gave up on hope about Nextcloud dev team add backup option

So I use now webdav with rclone

**Comment by schuc** [comment]
Over three year old request but I would also like to say that I would VERY much appreciate a one-way/backup option to/from Nextcloud for the Windows Desktop.

**Comment by EsmailELBoBDev2** [comment]
Right now you can use rclone with webdav until nextcloud dev pay attention to it's users

**Comment by flemmingss** [comment]
Well well, this is still wanted for me. Any plans?

**Comment by cbartondock** [comment]
Giga bump.

**Comment by whaddayakno** [comment]
bumpity bump

**Comment by tweak-wtf** [comment]
do the bump pls

**Comment by tio-trom** [comment]
I do not get why such file hosting platforms do not see this as a must feature...I could not find any such platform that has this. And it is sooo useful. Like for backups and such. At times because of timezone, server time, and other things, such platforms can be confused and try to download files to your machine again, thinking the ones in the nextcloud are newer.

In any case, a lot of uses for a one way sync. Please add this wonderful folks! :)

**Comment by meika** [comment]
maybe we should move this to reddit?

**Comment by MDE186** [comment]
Are there plans to implement this?

**Comment by EsmailELBoBDev2** [comment]
> Are there plans to implement this?

Use rclone with nextcloud as webdav

**Comment by ianp5a** [comment]
**Syncthing** software has the options **Up-Only, Down-only **and** Both-ways**. Those options would be very useful to have in the Nextcloud client. And shows those options are considered important.

**Comment by muava12** [comment]
> **Syncthing** software has the options:
> 
> * **Up-Only**
> * **Down-only**
> * **Both-ways**
> 
> Those options would be very useful to have in the Nextcloud client. And shows those options are considered important.

I'm just doing "send only", and delete from remote file.. annnnnddd, my files in client also deleta.. 😂. Yes its one way transfer but don't forget its also send 'deleted' files list

**Comment by danardi78** [comment]
My 10€ on this feature.
I'm unable to use webdav, sftp, etc.. only an option inside nextcloud clinet can save me <3

**Comment by paasi6666** [comment]
Any updates on this request?

**Comment by sebastianschauenburg** [comment]
I would like to kindly request to not bump or +1 this request, make a comment about on how much you would like this etc.. Please remember that every time anyone posts anything here, everyone subscribed will get a notification (in my case e-mail), that's why was even reluctant to post this reply (apologies to everyone in advance).

Let me be clear, I would also appreciate this feature (if it was built in). If it is created or someone contributes it, this issue will get updated. For now, this is not the case. Subscribe to this bug if you want to get notified if any updates occur in this issue.

Remark: you can of course also create this feature yourself and submit the patch upstream :wink: 

# workaround

For now I would like to mention an already shared workaround: [dav2fs](https://en.wikipedia.org/wiki/Davfs2) (or any other method to mount a [WebDAV](https://en.wikipedia.org/wiki/WebDAV) share). After the share is mounted on the filesystem, you can use any tool (even ```cp``` ) to make a 'one way sync'. Personally I leverage that WebDAV mountpoint with [rsync](https://en.wikipedia.org/wiki/Rsync) to sync data one way. Works like a charm.
